addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/protocol/KodiConnection.java

```diff
@@ -0,0 +1,571 @@
+                listener.updateAlbum(album);
+                listener.updateTitle(title);
+                listener.updateArtist(artist);
+                listener.updateMediaType(mediaType);
+            }
+
+        } catch (Exception e) {
+
+            logger.error("Event listener invoking error", e);
+        }
+    }
+
+    private JsonArray getJsonArray(String[] values) {
+        JsonArray result = new JsonArray();
+        for (String param : values) {
+            result.add(new JsonPrimitive(param));
+        }
+        return result;
+    }
+
+    private String convertFromArray(JsonArray data) {
+        StringBuilder result = new StringBuilder();
+        for (JsonElement x : data) {
+            if (result.length() > 0) {
+                result.append(", ");
+            }
+            result.append(convertToText(x));
+        }
+        return result.toString();
+    }
+
+    private String convertToText(JsonElement element) {
+        String text = element.getAsString();
+        try {
+            return new String(text.getBytes("ISO-8859-1"));
+        } catch (UnsupportedEncodingException e) {
+            return text;
+        }
+    }
+
+    private void updateState(KodiState state) {
+        // sometimes get a Pause immediately after a Stop - so just ignore
+        if (currentState.equals(KodiState.Stop) && state.equals(KodiState.Pause)) {
+            return;
+        }
+        try {
+            for (KodiEventListener listener : eventListeners) {
+                listener.updatePlayerState(state);
+                // if this is a Stop then clear everything else
+                if (state == KodiState.Stop) {
+                    listener.updateAlbum("");
+                    listener.updateArtist("");
+                    listener.updateTitle("");
+                    listener.updateMediaType("");
+                }
+            }
+
+        } catch (Exception e) {
+            logger.error("Event listener invoking error", e);
+        }
+
+        // keep track of our current state
+        currentState = state;
+    }
+
+    @Override
+    public void handleEvent(JsonObject json) {
+        JsonElement methodElement = json.get("method");
+
+        if (methodElement != null) {
+
+            String method = methodElement.getAsString();
+            JsonObject params = json.get("params").getAsJsonObject();
+            if (method.startsWith("Player.On")) {
+                processPlayerStateChanged(method, params);
+            } else if (method.startsWith("Application.On")) {
+                processApplicationStateChanged(method, params);
+            } else if (method.startsWith("System.On")) {
+                processSystemStateChanged(method, params);
+            } else if (method.startsWith("GUI.OnScreensaver")) {
+                processScreensaverStateChanged(method, params);
+            } else {
+                logger.warn("Received unknown method: {}", method);
```

Comment:
I think it is better so reduce the warning level to debug. When Kode updates its library e.g. there are a log of messages like this:
2016-11-26 11:33:36.524 [WARN ] [b.kodi.protocol.KodiConnection:341  ] - Received unknown method: AudioLibrary.OnUpdate

I have also seen several other methods that are not handled (and don't need to be handled). So I think level "debug" would be sufficient.